homework/App.js

```diff
@@ -7,72 +7,86 @@ class App {
     this.initialize(url);
   }
 
-  /**
-   * Initialization
-   * @param {string} url The GitHub URL for obtaining the organization's repositories.
-   */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { id: 'header' });
+    Util.createAndAppend('p', header, { text: 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, { id: 'select' });
+    const divGeneral = Util.createAndAppend('div', root, { id: 'div_general' });
+    const leftPanel = Util.createAndAppend('div', divGeneral, { id: 'left_panel' });
+    const leftPanelContainer = Util.createAndAppend('div', leftPanel);
+    const leftPanelContainerTable = Util.createAndAppend('table', leftPanelContainer);
+    const repositoryProperties = ['Repository :', 'Description :', 'Forks :', 'Updated :'];
+    const leftPanelContainerTableFirstTr = Util.createAndAppend('tr', leftPanelContainerTable);
+    Util.createAndAppend('td', leftPanelContainerTableFirstTr, {
+      text: repositoryProperties[0],
+    });
+    const linkRepo = Util.createAndAppend('td', leftPanelContainerTableFirstTr);
+    Util.createAndAppend('a', linkRepo, { id: 'property_repository' });
+    const rightPanel = Util.createAndAppend('div', divGeneral, { id: 'right_panel' });
+    Util.createAndAppend('p', rightPanel, { text: 'Contributions', id: 'right_column_title' });
+    const rigthColumnListDefault = Util.createAndAppend('ul', rightPanel, {
+      id: 'right_column_list',
+    });
 
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const data = await Util.fetchJSON(url);
+      this.data = data.map(repo => new Repository(repo));
+      this.data = data.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Hi Utku - there is something not right here. On line 34, you are creating a new repository object for each repository and assigning it to `this.data`. However, on the next line itself, you are trying to sort the fetched repositories data and assign it again to `this.data`.

This causes `this.data` to lose the repository objects you had created before. Due to this, you are having to use the `assignLeftPanelValues` method incorrectly at line 54 where you are using `.prototype` property. This method would have been already present with your object created in line 34 and you could have called it simply as `this.data[0].assignLeftPanelValues()`.

Could you please look into it? Drop me a message if you need any further information. 🙂

Reply:
Good evening Yash,

I stucked at this point.

I have tried to not touch skeleton code as much as I can when I have been writing this code.
But at this point, only solution that I can see is deleting `line 34`(it is skeleton). Because, I really bored with this homework. Struggled too much while trying to change code from one homework's objective to another. I felt that if i wrote whole code again.

I didn't understand classes(syntactic sugar/'salt') very well, I have tried to recover but for now I couldn't.

The problem is (from my side); there were no enough blueprints on JS3 HW1 & HW2 to achieve a code that can be easily(without major alteration) implemented one to another. In HW3, I faced a skeleton code that is totally conflicting with my code's layout. I have changed it as I can and it worked.

So, I don't have any idea about how I can transform existing code but i will try to focus again.

Reply:
I know that ignorance is not an option to become better but I believe that this homework does not have any benefit for me at this point, just wasting my time and confusing my brain. But as I said I will try.